jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -723,15 +724,12 @@ protected void acquire()
      * A variant of the {@link ArrayByteBufferPool} that
      * uses buckets of buffers that increase in size by a power of
      * 2 (e.g. 1k, 2k, 4k, 8k, etc.).
-     * @deprecated Usage of {@code Quadratic} is often wasteful of additional space and can increase contention on
```

Comment:
Interestingly enough, we have seen a few small contents being put into 4K buffers with the current buffer pool, so this pool may be more space efficient for those contents, as it can return a 1K buffer.

The main issue is for TLS, which needs an approx 17KB buffer, so it will get a 32KB one.

I wonder if it would be worthwhile to create a pool that wraps another pool and provides just a single specific sized buffer.   So we could have a pool specialized for TLS buffers wrapping a quadratic pool?  Best of both worlds?

Reply:
Should be easy to implement. Give it a go.
Or how about a dynamic one, that learns the sizes it needs to allocate?